simbad/command_line/simbad_database.py

```diff
@@ -163,6 +189,151 @@ def create_lattice_db(database):
     np.savez_compressed(database, niggli_data)
 
 
+def create_contaminant_db(database, nproc=2, submit_qtype=None, submit_queue=False):
+    """Create a contaminant database
+
+    Parameters
+    ----------
+    database : str
+       The path to the database folder
+    nproc : int, optional
+       The number of processors [default: 2]
+    submit_qtype : str
+       The cluster submission queue type - currently support SGE and LSF
+    submit_queue : str
+       The queue to submit to on the cluster
+
+    Raises
+    ------
+    RuntimeError
+        dimple.contaminants.prepare module not available
+    RuntimeError
+       Windows is currently not supported
+    """
+
+    try:
+        import dimple.contaminants.prepare
+    except ImportError:
+        msg = "This feature will be available with the next dimple update"
+        raise RuntimeError(msg)
+
+    if CUSTOM_PLATFORM == "windows":
+        msg = "Windows is currently not supported"
+        raise RuntimeError(msg)
+
+    dimple.contaminants.prepare.main(verbose=False)
+
+    simbad_dat_path = os.path.join(database, '*', '*', '*', '*.dat')
+    existing_dat_files = [os.path.basename(f).split('.')[0] for f in glob.glob(simbad_dat_path)]
+    erroneous_files = ['4V43']
+    dimple_files = ['cached', 'data.json', 'data.py']
+
+    with open("data.json") as data_file:
+        data = json.load(data_file)
+
+    results = []
+    for child in data["children"]:
+        try:
```

Comment:
Please put the `try - except` clause around the actual bit that raises it. Otherwise we might catch stuff we want to know about.

Reply:
The try - except clause is around the part that fails. There is an entry in child in data["children"] without the name "children", therefore if I put the try - except clause anywhere else I get a key error.